### ISSUE TITLE
TM_CURRENT_LINE can not get correct value

### ISSUE THREAD

**Issue by MoYuM** [ipatalas/vscode-postfix-ts#45]
I want to get current line code by [TM_CURRENT_LINE](https://code.visualstudio.com/docs/editor/userdefinedsnippets#_variables), but i get a wrong value

1. write a custom template like thie:
```
 {
            "name": "test",
            "body": "${TM_CURRENT_LINE}",
            "description": "test",
            "when": []
 }
```

 2. use this template with this code
```
someCode.test
```

3. result is
```
test
```

4. it should be 
```
someCode.test
```

**Comment by ipatalas** [comment]
Oh. Those templates were not available at the time I was referring to that syntax in README. Some of them seem to work, but not all of them.
Since it's not a classic VS Code snippet it won't work correctly.
To get the result you are expecting you could use this:
```JSON
{
    "name": "test",
    "body": "{{expr}}.test",
    "description": "test",
    "when": []
}
```
If that's not enough try explaining the exact thing you're trying to achieve.

**Comment by MoYuM** [comment]
I want to make a template to write `useState` in React, such like this
```
{
    "name": "useState",
    "body": "const [{{expr}}, set{{expr}}] = React.useState();",
    "description": "useState",
    "when": []
}
```

but i can not transform the first letter of `{{expr}}` to uppercase

the result is not good enough
```
const [count, setcount] = React.useState();
```

how can i make `setcount` to `setCount`?

**Comment by ipatalas** [comment]
Oh. I see. How was `TM_CURRENT_LINE` supposed to help here? Even if it worked correctly it would still have the same case.
I do have an idea though and it's not that difficult to implement.
One could just use `{{expr:lower}}`, `{{expr:upper}}`, `{{expr:title}}` so kind of filters for the replacement to give more flexibility.

**Comment by MoYuM** [comment]
`TM_CURRENT_LINE` will help me to get the code before dot.
like `{{expr}}`, but `{{expr}}` do not support [variable-transforms](https://code.visualstudio.com/docs/editor/userdefinedsnippets#_variable-transforms) which `TM_CURRENT_LINE` can.
then i can use regex to replace the first letter to uppercase, like [docs](https://code.visualstudio.com/docs/editor/userdefinedsnippets#_transform-examples) say.

but your filters is much better way to do such stuff. 

if you can support it , it will help a lot.

**Comment by ipatalas** [comment]
Ok, those transforms are pretty cool. TM_FILENAME works well even with the transforms but TM_CURRENT_LINE doesn't want to.
I guess it's because it's not a regular snippet. This is code based completion which partially supports snippets syntax but apparently not everything works yet.
As a workaround I can try with those filters I mentioned before. That will do the job for now. Maybe those snippets will start working one day, but I don't have time to deep dive into that now.

**Comment by MoYuM** [comment]
OK, thank you for your reply and your work of this stuff. it is very useful

**Comment by ipatalas** [comment]
Just released v1.9.4. Here is your template working:
```JSON
{
    "name": "useState",
    "body": "const [{{expr}}, set{{expr:capitalize}}] = React.useState();",
    "description": "const [{{expr}}, set{{expr:capitalize}}] = React.useState();",
    "when": []
},
```

**Comment by MoYuM** [comment]
Thank you !
